app/models/concerns/facets/managed_host_extensions.rb

```diff
@@ -0,0 +1,66 @@
+require 'facets'
+
+module Facets
+  module ManagedHostExtensions
+    extend ActiveSupport::Concern
+
+    included do
+      Facets::ManagedHostExtensions.refresh_facet_relations(self)
+
+      def attributes
+        hash = super
+
+        # include all facet attributes by default
+        facets_with_definitions.each do |facet, facet_definition|
+          hash["#{facet_definition.name}_attributes"] = facet.attributes.reject { |key, _| %w(created_at updated_at).include? key }
+        end
+        hash
+      end
+    end
+
+    def self.refresh_facet_relations(klass)
+      Facets.registered_facets.values.each do |facet_config|
+        self.register_facet_relation(klass, facet_config)
+      end
+    end
+
+    def self.register_facet_relation(klass, facet_config)
+      klass.class_eval do
+        has_one facet_config.name, :class_name => facet_config.model.name, :foreign_key => :host_id, :inverse_of => :host
+        accepts_nested_attributes_for facet_config.name
+        alias_method "#{facet_config.name}_attributes", facet_config.name
+
+        include facet_config.extension if facet_config.extension
+
+        include_in_clone facet_config.name
+
+        facet_config.compatibility_properties.each do |prop|
```

Comment:
How about extracting to a separate method to be called from `refresh_facet_relations`? That way you can add docs, override it if needed, etc.. Explain what compatibility_properties are